traffic_ops/traffic_ops_golang/trafficvault/backends/postgres/postgres.go

```diff
@@ -0,0 +1,233 @@
+}
+
+func (p *Postgres) DeleteOldDeliveryServiceSSLKeys(existingXMLIDs map[string]struct{}, cdnName string, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) GetCDNSSLKeys(cdnName string, tx *sql.Tx, ctx context.Context) ([]tc.CDNSSLKey, error) {
+	return nil, notImplementedErr
+}
+
+func (p *Postgres) GetDNSSECKeys(cdnName string, tx *sql.Tx, ctx context.Context) (tc.DNSSECKeysTrafficVault, bool, error) {
+	return tc.DNSSECKeysTrafficVault{}, false, notImplementedErr
+}
+
+func (p *Postgres) PutDNSSECKeys(cdnName string, keys tc.DNSSECKeysTrafficVault, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) DeleteDNSSECKeys(cdnName string, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) GetURLSigKeys(xmlID string, tx *sql.Tx, ctx context.Context) (tc.URLSigKeys, bool, error) {
+	return tc.URLSigKeys{}, false, notImplementedErr
+}
+
+func (p *Postgres) PutURLSigKeys(xmlID string, keys tc.URLSigKeys, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) GetURISigningKeys(xmlID string, tx *sql.Tx, ctx context.Context) ([]byte, bool, error) {
+	return nil, false, notImplementedErr
+}
+
+func (p *Postgres) PutURISigningKeys(xmlID string, keysJson []byte, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) DeleteURISigningKeys(xmlID string, tx *sql.Tx, ctx context.Context) error {
+	return notImplementedErr
+}
+
+func (p *Postgres) Ping(tx *sql.Tx, ctx context.Context) (tc.TrafficVaultPing, error) {
+	tvTx, dbCtx, cancelFunc, err := p.beginTransaction(ctx)
+	if err != nil {
+		return tc.TrafficVaultPing{}, err
+	}
+	defer p.commitTransaction(tvTx, dbCtx, cancelFunc)
+	n := 0
+	if err := tvTx.QueryRow("SELECT 1").Scan(&n); err != nil {
+		e := checkErrWithContext("Traffic Vault PostgreSQL: executing ping query", err, dbCtx.Err())
+		return tc.TrafficVaultPing{}, e
+	}
+	return tc.TrafficVaultPing{Status: "OK", Server: p.cfg.Hostname + ":" + strconv.Itoa(p.cfg.Port)}, nil
+}
+
+func (p *Postgres) GetBucketKey(bucket string, key string, tx *sql.Tx) ([]byte, bool, error) {
+	return nil, false, notImplementedErr
+}
+
+func init() {
+	trafficvault.AddBackend(postgresBackendName, postgresLoad)
+}
+
+func postgresLoad(b json.RawMessage) (trafficvault.TrafficVault, error) {
+	pgCfg := Config{}
+	if err := json.Unmarshal(b, &pgCfg); err != nil {
+		return nil, errors.New("unmarshalling Postgres config: " + err.Error())
+	}
+	if err := validateConfig(pgCfg); err != nil {
+		return nil, errors.New("validating Postgres config: " + err.Error())
+	}
+	if pgCfg.MaxIdleConnections == 0 {
+		pgCfg.MaxIdleConnections = defaultMaxIdleConnections
+	}
+	if pgCfg.ConnMaxLifetimeSeconds == 0 {
+		pgCfg.ConnMaxLifetimeSeconds = defaultConnMaxLifetimeSeconds
+	}
+	if pgCfg.QueryTimeoutSeconds == 0 {
+		pgCfg.QueryTimeoutSeconds = defaultDBQueryTimeoutSecs
+	}
```

Comment:
Instead of checking default values afterwards, can the defaults can be added to the initial value of `pgCfg` pre-unmarshal?